vars/createSha512.groovy

```diff
@@ -3,62 +3,24 @@ Closure call() {
 
     return { argsMap -> body: {
 
-        //for (file in findFiles(glob: "**/$argsMap.artifactPath/*")) {
-        //    acceptTypeFound = false
-        //    println("file: " + file)
+        final foundFiles = sh(script: "find $argsMap.artifactPath -type f", returnStdout: true).split()
 
-        //    for (fileType in allowedFileTypes) {
-        //        println("fileType: " + fileType)
-        //        if (file.endsWith(fileType)) {
-        //            sh("sha512sum ${file} > ${file}.sha512")
-        //            echo("Generating sha for ${file}")
-        //            acceptTypeFound = true
-        //            break
-        //        }
-        //    }
-
-        //    if (!acceptTypeFound) {
-        //        echo("Not generating sha for ${argsMap.artifactPath}, doesn't match allowed types ${allowedFileTypes}")
-        //    }
-        //}
-
-        sh """
-            set +x
-            file_type_list=".tar.gz .zip"
-            if [ -d "$argsMap.artifactPath" ]
-            then
-                for file_sub_path in `find $argsMap.artifactPath -type f`
-                do
-                    for file_type in \$file_type_list
-                    do
-                        if echo \$file_sub_path | grep -q \$file_type
-                        then
-                            echo Generating sha for \$file_sub_path
-                            sha512sum \$file_sub_path | while read sum filename
-                            do
-                                filename2=`basename \$filename`
-                                echo -e "\$sum \$filename2" > \$file_sub_path.sha512
-                            done
-                        fi
-                    done
-                done
-            else
-               for file_type in \$file_type_list
-               do
-                   if echo $argsMap.artifactPath | grep -q \$file_type
-                   then
-                       echo Generating sha for $argsMap.artifactPath
-                       echo Generating sha for $argsMap.artifactPath
-                       sha512sum $argsMap.artifactPath | while read sum filename
-                       do
-                           filename2=`basename \$filename`
-                           echo -e "\$sum \$filename2" > ${argsMap.artifactPath}.sha512
-                       done
-                   fi
-               done
-
-            fi
-        """
+        for (file in foundFiles) {
+            acceptTypeFound = false
+            for (fileType in allowedFileTypes) {
+                if (file.endsWith(fileType)) {
+                    echo("Creating sha for ${file}")
+                    final sha512 = sh(script: "sha512sum ${file}", returnStdout: true).split()
+                    // Reading the first index of array since sha512sum return [shasum, filename] as output
+                    writeFile file: "${file}_groovy_.sha512", text: "${sha512[0]}"
```

Comment:
The output of sha512sum has to be:
```
hashstring filename
```

Seems like you are only saying hashstring to the file?

Reply:
Yes, saving only the hash on the file. That's what we want right?

Reply:
no, we want hash then 2 spaces then only the basename of the file, see #1406.

Reply:
okay, I'll make the changes accordingly.